src/form/bf_mx_lin.jl

```diff
@@ -36,8 +36,24 @@ function constraint_mc_power_losses(pm::LPUBFDiagModel, n::Int, i, f_bus, t_bus,
     w_fr = var(pm, n, :w)[f_bus]
     w_to = var(pm, n, :w)[t_bus]
 
-    JuMP.@constraint(pm.model, p_fr + p_to .== diag( g_sh_fr).*w_fr + diag( g_sh_to).*w_to)
-    JuMP.@constraint(pm.model, q_fr + q_to .== diag(-b_sh_fr).*w_fr + diag(-b_sh_to).*w_to)
+
+    fb = ref(pm, n, :bus, f_bus)
+    tb = ref(pm, n, :bus, t_bus)
+    vmax1=fb["vmax"]
+    vmax2=tb["vmax"]
+
+    cnds = conductor_ids(pm; nw=n)
+    ncnds = length(cnds)
+	
+     for c in 1:ncnds
+     if((vmax1[c]!=0) && (vmax2[c]!=0))
+       JuMP.@constraint(pm.model, (p_fr[c] + p_to[c])*1000.0 == (g_sh_fr[c,c]*w_fr[c] +  g_sh_to[c,c]*w_to[c])*1000.0)
+       JuMP.@constraint(pm.model, (q_fr[c] + q_to[c])*1000.0 == (-b_sh_fr[c,c]*w_fr[c] + -b_sh_to[c,c]*w_to[c])*1000.0)
+    else
+       b=JuMP.@constraint(pm.model, (p_fr[c] + p_to[c])*1000.0 == 0)
+       JuMP.@constraint(pm.model, (q_fr[c] + q_to[c])*1000.0 == 0)
```

Comment:
I don't think the factors 1e3 should be needed here, they are on both sides and can be divided out cleanly